data/provider/src/main/java/tech/pegasys/teku/api/NodeDataProvider.java

```diff
@@ -13,14 +13,52 @@
 
 package tech.pegasys.teku.api;
 
+import java.util.List;
+import java.util.stream.Collectors;
+import tech.pegasys.teku.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.datastructures.operations.ProposerSlashing;
+import tech.pegasys.teku.datastructures.operations.SignedVoluntaryExit;
+import tech.pegasys.teku.statetransition.OperationPool;
 import tech.pegasys.teku.statetransition.attestation.AggregatingAttestationPool;
 
 public class NodeDataProvider {
 
   private final AggregatingAttestationPool attestationPool;
+  private final OperationPool<AttesterSlashing> attesterSlashingPool;
+  private final OperationPool<ProposerSlashing> proposerSlashingPool;
+  private final OperationPool<SignedVoluntaryExit> voluntaryExitPool;
 
-  public NodeDataProvider(final AggregatingAttestationPool attestationPool) {
+  public NodeDataProvider(
+      AggregatingAttestationPool attestationPool,
+      OperationPool<AttesterSlashing> attesterSlashingsPool,
+      OperationPool<ProposerSlashing> proposerSlashingPool,
+      OperationPool<SignedVoluntaryExit> voluntaryExitPool) {
     this.attestationPool = attestationPool;
+    this.attesterSlashingPool = attesterSlashingsPool;
+    this.proposerSlashingPool = proposerSlashingPool;
+    this.voluntaryExitPool = voluntaryExitPool;
+  }
+
+  public AggregatingAttestationPool getAttestationPool() {
+    return attestationPool;
+  }
```

Comment:
I don't think we should be exposing the `AggregatingAttestationPool` from here.

The way providers are currently setup, they never return internal objects, only the external ones.  We should follow that same pattern in this provider.

Reply:
Agreed, removed that function, it was not used anywhere.